Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -692,7 +715,7 @@ public static Branch getAutoInstance(@NonNull Context context) {
     public static Branch getAutoInstance(@NonNull Context context, boolean isReferrable) {
         isAutoSessionMode_ = true;
         customReferrableSettings_ = isReferrable ? CUSTOM_REFERRABLE_SETTINGS.REFERRABLE : CUSTOM_REFERRABLE_SETTINGS.NON_REFERRABLE;
-        boolean isDebug = BranchUtil.isTestModeEnabled(context);
+        boolean isDebug = BranchUtil.checkTestMode(context);
```

Comment:
does this `isDebug` flag imply the test is being used? If so, can you rename it to `isTestMode`? just to distinguish debug and test

Reply:
or should it be equal to `isDebugEnabled`?

Reply:
I named it `checkTestMode` because there is a side effect of saving a state variable.  `is` and `get` methods should be constant and have no side effects.

I couldn't think of any other descriptive prefaces though.  Semantically, this method's intention is to "check to see if test mode is enabled".   In this case it sounds better to just call it `isTestModeEnabled()`.

Reply:
Note that there is also a weird distinction between `isTestModeEnabled(context)` and `isTestModeEnabled()`.    These have different behaviors depending on the order called.

Reply:
Outdated